assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,679 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:EditBuildConfigController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('EditBuildConfigController', function ($scope, $routeParams, DataService, ProjectsService, BuildConfigsService, $filter, ApplicationGenerator, Navigate, $location, AlertMessageService, SOURCE_URL_PATTERN) {
+
+    $scope.projectName = $routeParams.project;
+    $scope.buildConfig = null;
+    $scope.alerts = {};
+    $scope.emptyMessage = "Loading...";
+    $scope.sourceURLPattern = SOURCE_URL_PATTERN;
+    $scope.options = {};
+    $scope.openshiftBuilderImages = ["ruby","python","perl","php","nodejs","wildfly"]
```

Comment:
Yeah wanted to go that way, but the problem was that most images(if they are custom) dont have annotation at all